test/tubes/serialtube_test.rb

```diff
@@ -0,0 +1,138 @@
+# encoding: ASCII-8BIT
+
+require 'open3'
+
+require 'test_helper'
+
+require 'pwnlib/tubes/serialtube'
+
+class SerialTest < MiniTest::Test
+  include ::Pwnlib::Tubes
+
+  def skip_windows
+    skip 'Not test tube/serialtube on Windows' if TTY::Platform.new.windows?
+  end
+
+  def open_pair
+    Open3.popen3('socat -d -d pty,raw,echo=0 pty,raw,echo=0') do |_i, _o, stderr, thread|
+      devs = []
+      2.times do
+        devs << stderr.readline.chomp.split.last
+        raise IOError, 'Could not create serial crosslink' if devs.last !~ %r{^/dev/pts/[0-9]+$}
```

Comment:
I tested on macOS, the output of socat is
```
➔ socat -d -d pty,raw,echo=0 pty,raw,echo=0
2018/06/10 13:44:41 socat[24931] N PTY is /dev/ttys005
2018/06/10 13:44:41 socat[24931] N PTY is /dev/ttys006
2018/06/10 13:44:41 socat[24931] N starting data transfer loop with FDs [5,5] and [8,8]
```
maybe that's the reason tests fails on travis-CI

Reply:
You nailed it. I had a strict regex filter that matched Linux style PTYs.
Thanks.